### PR TITLE
refactor some string operations with StringUtils

### PR DESCRIPTION
**Pull request by dj080808**
## What is the purpose of the change

refactor some string operations with StringUtils

## Brief changelog

XXXXX

## Verifying this change

XXXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

- [ ] Make sure there is a [GITHUB_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Format the pull request title like `[Dubbo-XXX] Fix UnknownException when host config not exist #XXX`. Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Run `mvn clean install -DskipTests=false` & `mvn clean test-compile failsafe:integration-test` to make sure unit-test and integration-test pass.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-common/src/main/java/org/apache/dubbo/common/URL.java [hunk 1]
```diff
@@ -1448,11 +1448,7 @@ public boolean equals(Object obj) {
...
         URL other = (URL) obj;
-        if (host == null) {
-            if (other.host != null) {
-                return false;
-            }
-        } else if (!host.equals(other.host)) {
+        if(!StringUtils.isEquals(host, other.host)) {
             return false;
         }
         if (parameters == null) {
```
Comment (biyuhao) [inline dubbo-common/src/main/java/org/apache/dubbo/common/URL.java]: what about this block, should we refactor at the same time?

Comment (dj080808) [reply dubbo-common/src/main/java/org/apache/dubbo/common/URL.java]: Should I split them into 2 different commits? Do you mean that?

Comment (biyuhao) [reply dubbo-common/src/main/java/org/apache/dubbo/common/URL.java]: oh, seems `parameters` is a Map, never mind.

**Comment by biyuhao** [comment]
LGTM

**Comment by kimmking** [comment]
well done

**Pull request merged**